### PR TITLE
Add check-onnx-backend to Mac CI.

### PR DESCRIPTION
**Pull request by etiotto**
Add check-onnx-backend to Mac CI.

Signed-off-by: Ettore Tiotto <etiotto@ca.ibm.com>

### CODE REVIEW

File: utils/check-onnx-numerical.cmd [hunk 1]
```diff
@@ -1,2 +1,2 @@
-cd onnx-mlir\build
+cd onnx-mlir/build
```
Comment (sstamenova) [inline utils/check-onnx-numerical.cmd:1]: Why is this changing?

File: CMakeLists.txt [hunk 2]
```diff
@@ -48,7 +48,13 @@ if (WIN32)
   # This allows us to pick up the correct version of Python instead of always picking the latest
   # installed on the machine.
   set(Python3_FIND_REGISTRY NEVER)
+  endif()
+
+# Use the Python version specified by pythonLocation.
+if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
```
Comment (sstamenova) [inline CMakeLists.txt:54]: Nit: space after if

File: CMakeLists.txt [hunk 3]
```diff
@@ -48,7 +48,13 @@ if (WIN32)
   # This allows us to pick up the correct version of Python instead of always picking the latest
   # installed on the machine.
   set(Python3_FIND_REGISTRY NEVER)
+  endif()
```
Comment (sstamenova) [inline CMakeLists.txt:51]: Nit: spacing

File: CMakeLists.txt [hunk 4]
```diff
@@ -48,7 +48,13 @@ if (WIN32)
...
+
+# Use the Python version specified by pythonLocation.
+if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
+  set(Python3_ROOT_DIR "$ENV{pythonLocation}")
```
Comment (sstamenova) [inline CMakeLists.txt:55]: Why is this needed? find_package below should find it and set the root dir, right?

**Review by sstamenova — CHANGES_REQUESTED** [review]

File: utils/check-onnx-numerical.cmd (hunk 1 above)
Comment (etiotto) [reply utils/check-onnx-numerical.cmd]: fixed

File: CMakeLists.txt (hunk 2 above)
Comment (etiotto) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 3 above)
Comment (etiotto) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 4 above)
Comment (sstamenova) [reply CMakeLists.txt:55]: So this change is needed just for the buildbot? Wouldn't it be better if we passed in the suggestion as a cmake variable?

e.g. `-DPython3_ROOT_DIR=...`

Comment (etiotto) [reply CMakeLists.txt:55]: That should work as well. The current code now allows for either passing `-DPython3_ROOT_DIR=...` via cmake or, if that is not defined, leveraging the `pythonLocation ` environment variable.  This is consistent with what is done by the third_party tools which has similar code: 

```
./onnx/third_party/pybind11/tools/pybind11NewTools.cmake:  if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
./pybind11/tools/pybind11NewTools.cmake:  if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
```

Comment (sstamenova) [reply CMakeLists.txt:55]: Reading environment variables like this is generally frowned upon in cmake. Unless there is a strong reason to do it, it is better to rely on the cmake options.

Comment (etiotto) [reply CMakeLists.txt]: ok done

**Review by sstamenova — APPROVED** [review]
Thanks!

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

**Comment by AlexandreEichenberger** [comment]
@etiotto Thanks for adding instructions in Readme. Ideally this one should go before #1084 as this info is moved around.
